### PR TITLE
search: Add screenreader only commas to query input suggestions

### LINKED ISSUE #43139: [Accessibility]: CodeMirror suggestions read all in one go and icons not read
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34188

### Problem description

CodeMirror suggestions are read all in one go, with no pause between the filter and the description. Also, the icons are not read; this is not a problem for filter suggestions as the type is in the description, but filter and repo suggestions don't 

image

### Expected behavior

- There should be some kind of pause between the name and the description (this can be achieved with a `<span className="sr-only">,</span>` I think)
- Non-symbol suggestions should have a type read (as represented by the icon; probably should use an `aria-label` on the icon). May want to do this for symbol suggestions too for consistency.

### Additional details

_No response_

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @lguychard

### PR DESCRIPTION
**Pull request by fkling**
Fixes #43139

This commit inserts screenreader-only commas between search suggestion parts.

## Test plan

Manual inspection of DOM. 

## App preview:

- [Web](https://sg-web-fkling-43139-vo-suggestions.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/search-ui/src/input/extensions/completion.ts [hunk 1]
```diff
@@ -146,16 +146,30 @@ export function searchQueryAutocompletion(
...
             render(completion) {
-                return createSVGIcon(
+                const icon = createSVGIcon(
                     completion.type && completion.type in typeIconMap
                         ? typeIconMap[completion.type as CompletionType]
                         : typeIconMap[SymbolKind.UNKNOWN]
                 )
+                if (completion.type && completion.type in typeIconMap) {
+                    icon.setAttribute('aria-label', completion.type)
+                }
```
Comment (limitedmage) [inline client/search-ui/src/input/extensions/completion.ts:156]: If there is no label, the icon should be hidden.

```suggestion
                if (completion.type && completion.type in typeIconMap) {
                    icon.setAttribute('aria-label', completion.type)
                } else {
                    icon.setAttribute('aria-hidden', 'true')
                }
```

FWIW no icon seems to have labels when I checked this out to test it? 
image

File: client/shared/src/util/dom.ts [hunk 2]
```diff
@@ -34,11 +34,10 @@ function elementIsContentEditable(element: HTMLElement): boolean {
 /**
  * Creates an SVG node. To be used together with path specs from @mdi/js
  */
-export function createSVGIcon(pathSpec: string): Node {
```
Comment (limitedmage) [inline client/shared/src/util/dom.ts:37]: Alternatively, add an optional label parameter here and, if not present, set aria-hidden to true. That would ensure that the icon always either has a label or is hidden.

**Review by limitedmage — APPROVED** [review]

File: client/shared/src/util/dom.ts (hunk 2 above)
Comment (fkling) [reply client/shared/src/util/dom.ts:37]: I like that!

**Comment by fkling** [comment]
@limitedmage I made some updates, could you test it again please?

**Review by limitedmage — APPROVED** [review]
Looks good! Still no labels for the icons but we can fix them later.

**Comment by fkling** [comment]
>  Still no labels for the icons but we can fix them later. 

Weird... it's definitely set on the elements 😞

**Pull request merged**